vm/systemSmartContracts/esdt.go

```diff
@@ -0,0 +1,279 @@
+//go:generate protoc -I=proto -I=$GOPATH/src -I=$GOPATH/src/github.com/gogo/protobuf/protobuf  --gogoslick_out=. esdt.proto
+package systemSmartContracts
```

Comment:
do not forget about the tests

Reply:
will come in next PR with other function implementations.